File: C#/03_write_to_shared_file.cs
using System;
using System.IO;
using System.Threading;

class Program
{
    static void WriteToFile(string fileName, int idx)
    {
        using (StreamWriter writer = File.AppendText(fileName))
        {
            writer.WriteLine("Thread{0} wrote to file", idx);
        }
    }

    static void Main(string[] args)
    {
        string fileName = "shared_file.txt";
        Thread[] threads = new Thread[10];

        for (int i = 0; i < 10; i++)
        {
            threads[i] = new Thread(() => WriteToFile(fileName, i));
            threads[i].Start();
        }

        for (int i = 0; i < 10; i++)
        {
            threads[i].Join();
        }
    }
}

File: python/03_write_to_shared_file.py
import threading

def write_to_file(file_name, data):
    with open(file_name, 'a') as f:
        f.write(data + '\n')

def main():
    file_name = 'shared_file.txt'
    threads = []

    for i in range(10):
        t = threading.Thread(target=write_to_file, args=(file_name, f'Thread {i} wrote to file'))
        threads.append(t)
        t.start()

    for t in threads:
        t.join()

if __name__ == '__main__':
    main()
